### PR TITLE
Migrate to be Ember Octane app

### PR DESCRIPTION
**Pull request by patocallaghan**
First off, apologies for the sheer size of this PR. I know this is potentially unreviewable but not sure of another way to do it 🙈

Some backstory. I wanted to implement an Octane version of the Ember Realworld application. I spoke to folks a while back and a separate application was suggested which was specifically based on the Octane edition, separate from the “Classic” Ember app. So I went off, and while I initially used the Ember Realworld fork as a base, I actually ended up rearchitecting and reimplementing a lot of what was in the Ember Classic application to follow a more component-driven architecture. This meant the codebase looked quite different than what was originally implemented. You can find that repo at [patocallaghan/ember-octane-realworld](https://github.com/patocallaghan/ember-octane-realworld).

I created an issue in Realworld for [Ember Octane](https://github.com/gothinkster/realworld/issues/449) and submitted this app and it was accepted. You can currently see it in the [framework list](https://github.com/gothinkster/realworld#frontends) and the actual app is running at https://ember-octane-realworld.netlify.com/.

image

It has since been suggested on Discord by @GCheung55 and @trek that this should actually be part of the main Ember Realworld because it’s still the same framework and just an update in syntax.

Conversation from Discord below:
image
image
image

So what do people think? cc @Alonski @mansona and any others who have opinions. If we’re happy to merge this in I’d then submit a PR back to RealWorld to remove the entry to the Octane specific version.

### CODE REVIEW

**Comment by Alonski** [comment]
Hmm this is very interesting. We could merge this in and then depreciate the Octane one.

@patocallaghan Matbe we can create an Octane branch here and split up this maybe PR into small ones? Then once that branch is ready we can merge into master?

File: .eslintrc.js [hunk 1]
```diff
@@ -1,30 +1,26 @@
...
     browser: true,
   },
   rules: {
-    'prettier/prettier': 'error',
-    'ember/order-in-components': 'error',
-    'ember/order-in-controllers': 'error',
-    'ember/order-in-routes': 'error',
-    'prefer-const': 'error',
-    'no-var': 'error',
+    'ember/use-ember-data-rfc-395-imports': 'error',
```
Comment (Alonski) [inline .eslintrc.js:17]: Is there a reason why `prettier/prettier`, `prefer-const` and `no-var` were removed?

File: .prettierrc.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
-printWidth: 120
+printWidth: 100
```
Comment (Alonski) [inline .prettierrc.yml:1]: Any reason for this?
Personally I have always used 120 that is why this was at 120. 
Is this just a personal preference?

Comment (patocallaghan) [reply .prettierrc.yml:1]: Yeah no reason other than I think this the default in Prettier and what I had in the Octane repo. Happy to change back.

File: .eslintrc.js (hunk 1 above)
Comment (patocallaghan) [reply .eslintrc.js:17]: I've simply used Ember CLI's 3.16 default eslintrc.js and just added the prettier and ember-data rules.

**Comment by Alonski** [comment]
@patocallaghan Can you try to get Netlify deploy preview working?
After that lets just merge this in!

**Pull request merged**